resolwe/test/utils.py

```diff
@@ -66,26 +71,83 @@ def check_docker():
     return available, reason
 
 
-def with_docker_executor(method):
-    """Decorate unit test to run processes with Docker executor."""
+def with_custom_executor(wrapped=None, **custom_executor_settings):
```

Comment:
I agree, this is a good idea.

> Now only test methods can be decorated, right?

Yes, this is a limitation of the current decorator.

> Is it a lot of work to support this?

I don't know, I'll check up how much work it would be.

For the time being, I would like to merge this PR as is and then iterate on these kinds of improvements.